src/extensions/sessionrecording-utils.js

```diff
@@ -34,3 +36,39 @@ export function filterDataURLsFromLargeDataObjects(data) {
     }
     return data
 }
+
+export const CONSOLE_LOG_PLUGIN_NAME = 'rrweb/console@1' // The name of the rr-web plugin that emits console logs
+
+// Console logs can be really large. This function truncates large logs
+// It's a simple function that just truncates long strings.
+// TODO: Ideally this function would have better handling of objects + lists,
+// so they could still be rendered in a pretty way after truncation.
+export function truncateLargeConsoleLogs(event) {
+    const MAX_STRING_SIZE = 2000 // Maximum number of characters allowed in a string
+    const MAX_STRINGS_PER_LOG = 10 // A log can consist of multiple strings (e.g. consol.log('string1', 'string2'))
+
+    if (
+        event &&
+        typeof event === 'object' &&
```

Comment:
curious: why do you need to check the types here?

or, what other kinds of events are we expecting in here?

Reply:
It _should_ always be an object, but I try to be safe when parsing rr-web events. These events come from the browser's MutationObserver -> RRWeb -> us, so there's plenty of room for unexpected behavior.